### PR TITLE
CMR-8843: Rename association-details to association_details, concept-id to concept_id

### PR DESCRIPTION
**Pull request by eudoroolivares2016**

### CODE REVIEW

File: search-app/src/cmr/search/results_handlers/results_handler_util.clj [hunk 1]
```diff
@@ -1,8 +1,17 @@
...
   (:require
+   [clojure.walk :as walk]
    [cmr.common.util :as util]))
 
+(defn replace-snake-keys
+ "Replace specified keys in the nested association data structure with
+ snake_case version"
+ [associations]
+ (->> associations
+   (walk/postwalk-replace {:concept-id :concept_id})
```
Comment (daniel-zamora) [inline search-app/src/cmr/search/results_handlers/results_handler_util.clj:12]: alignment off

File: search-app/test/cmr/search/test/results_handlers/results_handler_util.clj [hunk 2]
```diff
@@ -58,3 +59,15 @@
...
             {}
             :collection))))
+
+(comment "Useful for debugging nested association DS"
+  ((def assoc {:variables [{:concept-id "V1200000019-PROV1"} {:concept-id "V1200000021-PROV1"}]
+                      :services [{:concept-id "S1200000009-PROV1" :data {:convert-format {:XYZ "ZYX"}, :allow-regridding "true"}}]
+                      :tools [{:concept-id "TL1200000010-PROV1"}]
+                      :data-quality-summaries [{:concept-id "DQS1200000012-PROV1", :data {:XYZ "ZYX"}, :revision-id 1}],
+                      :order-options [{:concept-id "OO1200000014-PROV1"} {:concept-id "OO1200000015-PROV1"}]})
+    def assocVals (vals {:variables [{:concept-id "V1200000019-PROV1"} {:concept-id "V1200000021-PROV1"}]
```
Comment (daniel-zamora) [inline search-app/test/cmr/search/test/results_handlers/results_handler_util.clj:69]: missing `(`, maybe above?

File: search-app/test/cmr/search/test/results_handlers/results_handler_util.clj [hunk 3]
```diff
@@ -58,3 +59,15 @@
             nil
             {}
             :collection))))
+
+(comment "Useful for debugging nested association DS"
+  ((def assoc {:variables [{:concept-id "V1200000019-PROV1"} {:concept-id "V1200000021-PROV1"}]
```
Comment (daniel-zamora) [inline search-app/test/cmr/search/test/results_handlers/results_handler_util.clj:64]: `((`

File: system-int-test/test/cmr/system_int_test/search/generic_association_test.clj [hunk 4]
```diff
@@ -51,9 +51,11 @@
                        (get-in (first (:items search-body)) [:associations concept-type-plural]))
         association-details (if meta?
                               (get-in (first (:items search-body)) [:meta :association-details concept-type-plural])
-                              (get-in (first (:items search-body)) [:association-details concept-type-plural]))]
+    ;; association_details (snake_case) is used for the .json endpoit which is being called if meta = false
```
Comment (daniel-zamora) [inline system-int-test/test/cmr/system_int_test/search/generic_association_test.clj:54]: `endpoit`
Also I think it need to be aligned with the line under it.

File: system-int-test/test/cmr/system_int_test/search/generic_association_test.clj [hunk 5]
```diff
@@ -248,9 +250,10 @@
         (is (= [tl1-concept-id]
                (:associations grid-search-result)
                (:associations grid-search-result-update)))
-        (is (= [{:concept-id tl1-concept-id :revision-id tl1-revision-id}]
-               (:association-details grid-search-result)
-               (:association-details grid-search-result-update)))
+        (is (= [{:concept_id tl1-concept-id :revision_id tl1-revision-id}]
+        ;; The grid is being searched from the .json endpoint which uses snake case assocation-details
```
Comment (daniel-zamora) [inline system-int-test/test/cmr/system_int_test/search/generic_association_test.clj:254]: check with the style guide, but I think this needs to be aligned with code below.

File: search-app/docs/api.md [hunk 6]
```diff
@@ -5410,9 +5410,9 @@ Example un-healthy response body:
...
- 
-#### <a name="concept-associations"></a> Concept associations 
+A new association API has been developed to achieve the goal of being able to associate a concept of any type, with or without revision, to one or more other concepts of any type, with or without revisions. The new association API also allows the associations to include an optional data payload, whose purpose is to describe the association itself. Associations which do not initially have an association data payload, may have it added through an association update. A concept can only be associated with another concept either with or without revisions, not both. A concept can not be associated to itself, even with different revisions. It's worth noting that associations between collections and services/tools/variables cannot be made through the new association API because these associations require different business rules. These existing associations will continue to be made through the existing association API.
```
Comment (isja17) [inline search-app/docs/api.md:5413]: is there any way we can split this paragraph up? It's information dense, and throws a lot at the user all at once.

**Review by ygliuvt — APPROVED** [review]

File: search-app/docs/api.md (hunk 6 above)
Comment (eudoroolivares2016) [reply search-app/docs/api.md:5413]: That is a very good point I think I could split it into two section: The top level information about the generic association API and then another section with something along the lines of "important-notes" that breakdown some of the subtleties of what can be done,

File: search-app/test/cmr/search/test/results_handlers/results_handler_util.clj (hunk 2 above)
Comment (eudoroolivares2016) [reply search-app/test/cmr/search/test/results_handlers/results_handler_util.clj]: Removed and adjusted comment section

File: system-int-test/test/cmr/system_int_test/search/generic_association_test.clj (hunk 5 above)
Comment (eudoroolivares2016) [reply system-int-test/test/cmr/system_int_test/search/generic_association_test.clj]: Adjusting comments for indentation

File: search-app/src/cmr/search/results_handlers/results_handler_util.clj (hunk 1 above)
Comment (eudoroolivares2016) [reply search-app/src/cmr/search/results_handlers/results_handler_util.clj]: Fixed alignment for function

**Review by isja17 — APPROVED** [review]

**Review by zimzoom — APPROVED** [review]

**Pull request merged**